src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -732,11 +732,60 @@ export function appendComponentTypeForFormatToTextureType(base: string, format:
     : `${base}<${getTextureFormatTypeInfo(format).componentType}>`;
 }
 
-function createRandomTexelViewViaColors(info: {
-  format: GPUTextureFormat;
-  size: GPUExtent3D;
-  mipLevel: number;
-}): TexelView {
+type RandomTextureOptions = {
+  generator: PerPixelAtLevel<PerTexelComponent<number>>;
+};
+
+/**
+ * Make a generator for texels for depth comparison tests.
+ */
+export function makeRandomDepthComparisonTexelGenerator(
+  info: {
+    format: GPUTextureFormat;
+    size: GPUExtent3D;
+  },
+  comparison: GPUCompareFunction
+) {
+  const rep = kTexelRepresentationInfo[info.format as EncodableTextureFormat];
+  const size = reifyExtent3D(info.size);
+
+  const isEqualNotEqual = comparison === 'equal' || comparison === 'not-equal';
+
+  // for equal and not-equal we just want to test 0, 0.5, and 1
+  // for everything else we want 0 to 1
+  const fixedValues = [0, 0.5, 1, 1];
+  const format = isEqualNotEqual
+    ? (norm: number) => fixedValues[(norm / 3) | 0]
```

Comment:
norm will be random number number between [0, 1], This will only return fixedValues[0]?
I think here you want something like `fixedValues[(norm / 0.3333333) | 0]`?
more formally something like  `fixedValues[Math.floor(norm * fixedValues.length)]`

Reply:
`fixedValues[Math.floor(norm * fixedValues.length)]` doesn't work because `norm` goes from 0 to 1, not 0 to 0.99999999999 so when norm is 1 it will pull the last number. 

I changed it to `fixedValues[(norm * (fixedValues.length - 1)) | 0]`